tests/generation/test_utils.py

```diff
@@ -1832,18 +1832,15 @@ def test_generate_from_inputs_embeds_decoder_only(self):
     def test_generate_continue_from_past_key_values(self):
         # Tests that we can continue generating from past key values, returned from a previous `generate` call
         for model_class in self.all_generative_model_classes:
-            # won't fix: old models with unique inputs/caches/others
             if any(model_name in model_class.__name__.lower() for model_name in ["imagegpt"]):
-                return
-            # may fix in the future: needs modeling or test input preparation fixes for compatibility
+                self.skipTest("Won't fix: old model with unique inputs/caches/other")
             if any(model_name in model_class.__name__.lower() for model_name in ["umt5"]):
-                return
+                self.skipTest("TODO: needs modeling or test input preparation fixes for compatibility")
```

Comment:
This is better, but as we are in `for model_class in self.all_generative_model_classes`, it probably means we skip many `model_class` unnecessary if a previous one is skipped here.

Good to merge. I can do a follow up PR to use `subTest` (as an exercise for me)

Reply:
After a second look, subTest doesn't make much sense here.